services/calculate.go

```diff
@@ -66,38 +65,49 @@ func CalculateOverallScore(scoresURL string) (*models.ScoreResponse, error) {
 	if port == "" {
 		port = getDefaultPort(protocol)
 	}
-	protocolScore, protocolMessage := CalculateProtocolScore(protocol)
-	messages = append(messages, protocolMessage)
+	protocolScore := CalculateProtocolScore(protocol)
 	score = score + protocolScore
 	var maximumScore = 5
-	headerScore, _, maxScore, ServerDetail, err := GetResponseHeaderScore(scoresURL)
+	headerScore, _, maxScore, ServerDetail, ServerData, err := GetResponseHeaderScore(scoresURL)
 	if err != nil {
-		fmt.Println("hello", err)
 		return nil, err
 	}
 	maximumScore = maximumScore + maxScore
 	score = score + headerScore
-	mailServerScore, maxScore := GetMailServerConfigurationScore(host)
+	mailServerScore, maxScore, txtRecords, dmarcRecords := GetMailServerConfigurationScore(host)
 	score += mailServerScore
 	maximumScore += maxScore
 	totalScore := math.Ceil((float64(float64(score)/float64(maximumScore)))*100) / 100
-	fmt.Println("Protocol Score is " + strconv.Itoa(protocolScore))
-	fmt.Println("Message: " + protocolMessage)
 	fmt.Println("Final Score for: " + scoresURL + " is " + strconv.Itoa(score) + " out of " + strconv.Itoa(maximumScore))
 	certificates, certError := models.GetCertificate(host, port, protocol)
 	if certError != nil {
 		return nil, certError
 	}
-	scores := models.GetScores(scoresURL, totalScore, messages)
+	scores := models.GetScores(scoresURL, totalScore, badges)
 	response := models.GetScoresResponse(scores, certificates, nil, ServerDetail)
-	return response, nil
+	responseBody, err := json.Marshal(response)
+	serverdataJSON, serverdataJSONerr := json.Marshal(ServerData)
+	if serverdataJSONerr != nil {
+		fmt.Println("Error Occured while parsing Server Data JSON", serverdataJSONerr)
+	}
+	entry := &models.Domain{
+		Name:         scoresURL,
+		ServerData:   string(serverdataJSON),
+		TxtRecords:   txtRecords,
+		DmarcRecords: dmarcRecords,
+		Response:     string(responseBody),
+		IncidentList: "",
+		Score:        totalScore,
+	}
+	utils.CreateEntry(entry)
+	return responseBody, err
 }
 
 // GetResponseHeaderScore returns the Response Header Score for the HTTP Request
-func GetResponseHeaderScore(url string) (totalScore int, XSSReportURL string, maxScore int, serverInfo *models.ServerDetail, err error) {
+func GetResponseHeaderScore(url string) (totalScore int, XSSReportURL string, maxScore int, serverInfo *models.ServerDetail, serverData map[string]string, err error) {
```

Comment:
This methods args seem to be long. It might be better to encapsulate some args into a struct maybe. Its a good indicator to avoid refactor func/methods that take more than 3 args.

Reply:
This isnt mandatory as such, Ill try to add in those changes later.